benchmarks/dbscan/data.hpp

```diff
@@ -26,19 +26,32 @@ std::vector<Point<DIM>> sampleData(std::vector<Point<DIM>> const &data,
 {
   std::vector<Point<DIM>> sampled_data(num_samples);
 
-  std::default_random_engine generator(17);
-  // FIXME_NVCC(11.0-11.3): not providing int here results in an internal
-  // compiler error
-  std::uniform_int_distribution<int> distribution;
-  auto rand = [&distribution, &generator]() { return distribution(generator); };
+  // C++ standard does not guarantee portability of <random>. While by some
+  // luck random engines are portable (except for the `default_random_engine`),
+  // `uniform_int_distribution` is not.
+  //
+  // The problem is then that when we sample a dataset on different
+  // platforms/compilers, we get a different data. It is problematic for
+  // reproduciblity.
+  //
+  // Thus, we use a hardcoded Lehmer (or Park-Miller) random generator in
+  // place, as it is fast, easy to implement, and does what we need.
+
+  // The state must be initialized to any positive number less than modulus,
+  // which is num_samples here
+  assert(num_samples > 1);
+  auto rand = [state = (1337 % (num_samples - 1) + 1)]() mutable {
+    state = ((unsigned long long)state * 48271) % 0x7fffffff;
```

Comment:
What is suboptimal about that code+comment is that one cannot tell apart the magic numbers commonly used, the `48271` parameter ("multiplier") and the `0x7fffffff` hex number ("prime modulus" which is 2^31-1), from the seed being picked up.